packages/file_selector/file_selector_web/lib/src/dom_helper.dart

```diff
@@ -3,41 +3,46 @@
 // found in the LICENSE file.
 
 import 'dart:async';
-import 'dart:html';
+import 'dart:js_interop';
 
 import 'package:file_selector_platform_interface/file_selector_platform_interface.dart';
 import 'package:flutter/foundation.dart' show visibleForTesting;
 import 'package:flutter/services.dart';
+import 'package:web/helpers.dart';
 
 /// Class to manipulate the DOM with the intention of reading files from it.
 class DomHelper {
   /// Default constructor, initializes the container DOM element.
   DomHelper() {
     final Element body = querySelector('body')!;
-    body.children.add(_container);
+    body.appendChild(_container);
   }
 
-  final Element _container = Element.tag('file-selector');
+  final Element _container = createElementTag('file-selector');
 
   /// Sets the <input /> attributes and waits for a file to be selected.
   Future<List<XFile>> getFiles({
     String accept = '',
     bool multiple = false,
-    @visibleForTesting FileUploadInputElement? input,
+    @visibleForTesting HTMLInputElement? input,
   }) {
     final Completer<List<XFile>> completer = Completer<List<XFile>>();
-    final FileUploadInputElement inputElement =
-        input ?? FileUploadInputElement();
+    final HTMLInputElement inputElement =
+        input ?? (createElementTag('input') as HTMLInputElement)
+          ..type = 'file';
 
```

Comment:
It'd be nice if this asserted that `input.type == 'file'`, to retain some of the old type safety.

Reply:
not worried about this, since it's text only